src/algebra/Mon/basic.lean

```diff
@@ -33,6 +33,14 @@ abbreviation forget : Mon.{u} ⥤ Type u := forget
 
 instance hom_is_monoid_hom {R S : Mon} (f : R ⟶ S) : is_monoid_hom (f : R → S) := f.2
 
+/-- Morphisms in `Mon` are defined using `subtype is_monoid_hom`,
+so we provide a canonical bijection with `R →* S`. -/
+def hom_equiv_monoid_hom (R S : Mon) : (R ⟶ S) ≃ (R →* S) :=
```

Comment:
This should be a definitional equality I think. One of the definitions should be changed to make it defeq to the other.

Reply:
Of course, you're right. I'm going to migrate `concrete_category` and all instances (`Mon`, `Group` etc) to bundled morphisms but I'd prefer to have this PR merged before that. I can add a comment saying that this is a temporary `equiv` that will be deleted once `Mon` use `→*` as morphisms.